EAP-TLS_Solution/EAP-TLS WebAPI (Core)/README.md

```diff
@@ -0,0 +1,94 @@
+# EAP-TLS solution: WebAPI Server
+
+This is an EAP-TLS WebAPI Server solution, developed on top of the [ASP.NET Core Framework 3.1](https://docs.microsoft.com/en-us/aspnet/core/?view=aspnetcore-3.1), which is "acting-as" the publicly-exposed endpoint to which Clients (i.e. Azure Sphere devices) can connect for their first provisioning and/or certificate renewals. It substantially mimics an MdM Server, and it's internals showcase to developers how to extract Azure Sphere's DAA certificate directly from the TLS handshaking, without any pre-provisioning.
+
+## Prerequisites
+
+- The web server can be hosted on a dedicated machine, or self-hosted for easy development purposes. When self-hosted, the WebAPI's endpoint, `api/certs`, will be located at the Host machine's IP Address, typically on port 44378 (i.e. `https://192.168.0.2:44378/api/certs`, if the development Host machine's IP address is `192.168.0.2`).
+- The Host machine must be accessible from the devices that will be running the [Azure Sphere EAP-TLS&#32;Client](..\EAP-TLS&#32;Client) solution.
+- A test-WebAPI private certificate, in **PFX** format (i.e. `eap-tls-webapi.pfx`).
+- A test-RADIUS RootCA public certificate, in **PEM** format (i.e. `az-CA.pem`), for authenticating RADIUS Server.
+- A test-RADIUS Client private certificate, split in two different files, both in **PEM** format: the *public* portion (i.e. `iotuser_public.pem`) and the *private key* portion (i.e. `iotuser_private.pem`).
+
+  **Note**: if you don't have certificates, follow the steps in [How to generate certificates for testing](https://github.com/Azure/azure-sphere-samples/tree/master/Samples/Certificates/Cert_HighLevelApp/get-certificates.md) to create the certificates.
+
+## Configuring the WebAPI Server
+
+For demo purposes, the WebAPI Server handles just one Client, while a full solution should rely on a proper CMS (Certificate Management System) to identify multiple incoming device IDs, and issue the corresponding certificates and identities. See the WebAPI's controller source code ([CertsController.cs](.\/Controllers/CertsController.cs)) for full details and recommendations within the comments.
+
+Make sure to:
+
+- Copy th following certificates in the [EAP-TLS&#32;WebAPI&#32;(Core)\certs](.\EAP-TLS&#32;WebAPI&#32;(Core)/certs) folder:
+  - The WebAPI's private certificate, in PFX format (i.e. `eap-tls-webapi.pfx`).
+  - The RADIUS public RootCA certificate, in PEM format (i.e. `az-CA.pem`).
+  - The RADIUS public Client certificate and related private key, both in PEM format (i.e. `iotuser_public.pem`).
+- In order for the DAA process to succeed, on the machine in which the WebAPI will be running, you must download and register (*) the public certificate of the AS3 Tenant into which the Azure Sphere device was claimed in:
```

Comment:
Grammar nit:  just "into which the Azure Sphere device was claimed:  (i.e. drop the "in").